libs/native/src/main22/java/org/elasticsearch/nativeaccess/jdk/ArenaUtil.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.nativeaccess.jdk;
+
+import java.lang.foreign.Arena;
+import java.lang.foreign.MemoryLayout;
+import java.lang.foreign.MemorySegment;
+import java.nio.charset.Charset;
+
+public class ArenaUtil {
+
+    /**
+     * Allocate an array of the given memory layout.
+     */
+    static MemorySegment allocate(Arena arena, MemoryLayout layout, int count) {
+        return arena.allocate(layout, count);
```

Comment:
Just curiosity: this is due to an API change from 21 to 22, correct?

Reply:
Yes, allocateArray was renamed